src/quick-lint-js/parse.h

```
@@ -3006,6 +3006,19 @@ class parser {
           break;
         }
 
+        case token_type::kw_new: {
```

We should also report an error for other tokens, such as `class` and `typeof`.